tsconfig.json

```diff
@@ -7,7 +7,7 @@
     "inlineSources": true,
     "importHelpers": true,
     "lib": ["es6"],
-    "module": "commonjs",
+    "module": "es2020",
```

Comment:
Can you keep the top level `tsconfig.json` unchanged cause it's like a template of all packages. It should be enough to change package level `tsconfig.json`.

Reply:
the [typescript rollup plugin](https://github.com/rollup/plugins/tree/master/packages/typescript#options) in `rollup.config.js` overrides options in tsconfig. So if `rollup.config.js` is changed to use es2020 [here](https://github.com/amplitude/Amplitude-TypeScript/pull/983#discussion_r1983750113). It doesn't matter which module version we use here.